### PR TITLE
[Redis 6.2] Add idle option to XPENDING

### PR DESCRIPTION
**Pull request by a0n**
Added support for [Idle time filter](https://redis.io/commands/xpending/#idle-time-filter) on the XPENDING command, which is available since 6.2.0.

Reference: https://github.com/redis/redis-rb/issues/978

### CODE REVIEW

File: lib/redis/commands/streams.rb [hunk 1]
```diff
@@ -344,10 +346,13 @@ def xautoclaim(key, group, consumer, min_idle_time, start, count: nil, justid: f
...
       # @param consumer [String]  the consumer name
       #
+      # @option opts [Integer] :idle       pending message minimum idle time in milliseconds
+      #
       # @return [Hash]        the summary of pending entries
       # @return [Array<Hash>] the pending entries details if options were specified
-      def xpending(key, group, *args)
+      def xpending(key, group, *args, **opts)
         command_args = [:xpending, key, group]
+        command_args.concat(['IDLE', opts[:idle].to_i])  if opts[:idle]
```
Comment (byroot) [inline lib/redis/commands/streams.rb:355]:
```suggestion
      def xpending(key, group, *args, idle: nil)
        command_args = [:xpending, key, group]
        command_args << 'IDLE' << Integer(idle) if idle
```

File: test/lint/streams.rb [hunk 2]
```diff
@@ -752,6 +752,49 @@ def test_xpending_with_range_options
       assert_equal 1, actual[2]['count']
     end
 
+    def test_xpending_with_range_and_idle_options
+      redis.xadd('s1', { f: 'v1' }, id: '0-1')
+      redis.xgroup(:create, 's1', 'g1', '$')
+      redis.xadd('s1', { f: 'v2' }, id: '0-2')
+      redis.xadd('s1', { f: 'v3' }, id: '0-3')
+      redis.xreadgroup('g1', 'c1', 's1', '>')
+
```
Comment (casperisfine) [inline test/lint/streams.rb:761]: Rubocop is unhappy
```
Offenses:

test/lint/streams.rb:761:1: C: [Correctable] Layout/TrailingWhitespace: Trailing whitespace detected.
```

File: test/lint/streams.rb [hunk 3]
```diff
@@ -752,6 +752,49 @@ def test_xpending_with_range_options
       assert_equal 1, actual[2]['count']
     end
 
+    def test_xpending_with_range_and_idle_options
+      redis.xadd('s1', { f: 'v1' }, id: '0-1')
```
Comment (casperisfine) [inline test/lint/streams.rb:756]: This test needs to be wrapped in a `target_version "6.2" do` block.

**Review by casperisfine — COMMENTED** [review]
Also please squash your commits.

**Comment by a0n** [comment]
Thanks for the Feedback and the Merge.

**Pull request closed without merging**